### PR TITLE
Add config file support for existing parameters

### LINKED ISSUE #403: Add Config File Support for Existing Parameters
# 💡 Summary #
Add the capability to specify the existing ScubaGoggles parameters via a config file.

## Motivation and context ##
See https://github.com/cisagov/ScubaGoggles/issues/161.

## Implementation notes ##
See https://github.com/cisagov/ScubaGear/tree/main/PowerShell/ScubaGear/Sample-Config-Files

## Acceptance criteria ##
- [ ] Existing parameters can be input via config file
- [ ] Plan for what happens when a parameter is supplied via both the command line and config file

### PR DESCRIPTION
**Pull request by adhilto**
## 🗣 Description ##
Added config file support for the existing ScubaGoggles parameters. See https://github.com/cisagov/ScubaGoggles/blob/403-add-config-file-support-for-existing-parameters/docs/usage/Config.md for usage details.

### 💭 Motivation and context
Closes #403.

## 🧪 Testing
Note this adds a new dependency: pyyaml. Be sure to install that before testing.

Manually tested:
- Ran without config file
- Tested with config file
- Tested with parameters in both config file and specified in the command-line
- Tested with a mix of both full parameter names and aliases (i.e., `-b` vs `--baselines`)

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] If applicable, *All* future TODOs are captured in issues, which are referenced in the PR description.
- [x] The relevant issues PR resolves are linked preferably via [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [x] All relevant type-of-change labels have been added.
- [x] I have read and agree to the [CONTRIBUTING.md](https://github.com/cisagov/ScubaGoggles/blob/main/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge Checklist

- [ ] This PR has been smoke tested to ensure main is in a functional state when this PR is merged.
- [ ] Squash all commits into one PR level commit using the `Squash and merge` button.

## ✅ Post-merge Checklist

- [ ] Delete the branch to clean up.
- [ ] Close issues resolved by this PR if the [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) did not activate.

### CODE REVIEW

File: sample-config-files/basic_config.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# YAML basic configuration file with examples of how to specify various types
+# of parameters.
+
+# Example specifying a string parameter
+outputpath: example_output
+
+# Example specifying a list parameter
+baselines: [gmail, calendar, groups, chat, drive, meet, sites, commoncontrols]
```
Comment (mitchelbaker-cisa) [inline sample-config-files/basic_config.yaml:8]: It'd be helpful to know if arguments need to be denoted as a string or list type. 

For example the following will fail: 
```
baselines: gmail, calendar, groups, chat, drive, meet, sites, commoncontrols
or
b: *
```

**Review by mitchelbaker-cisa — APPROVED** [review]
Looks good, Tested with/without a config file when run via scubagoggles gws and scuba.py. Tested with parameters in both config file/command-line, and with a mix of full param names/aliases.

Left one comment for consideration.

File: sample-config-files/basic_config.yaml (hunk 1 above)
Comment (adhilto) [reply sample-config-files/basic_config.yaml:11]: Good point. I added a comment that hopefully clarifies it.

**Review by LaurenBassett — APPROVED** [review]

**Comment by LaurenBassett** [comment]
Tests ran fine for me :-)

**Pull request merged**